spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/StreamDeploymentPropertiesUtils.java

```diff
@@ -0,0 +1,306 @@
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.cloud.dataflow.configuration.metadata.ApplicationConfigurationMetadataResolver;
+import org.springframework.cloud.dataflow.core.ApplicationType;
+import org.springframework.cloud.dataflow.core.BindingPropertyKeys;
+import org.springframework.cloud.dataflow.core.DataFlowPropertyKeys;
+import org.springframework.cloud.dataflow.core.StreamAppDefinition;
+import org.springframework.cloud.dataflow.core.StreamDefinition;
+import org.springframework.cloud.dataflow.core.StreamPropertyKeys;
+import org.springframework.cloud.dataflow.registry.AppRegistration;
+import org.springframework.cloud.dataflow.registry.AppRegistry;
+import org.springframework.cloud.dataflow.rest.util.DeploymentPropertiesUtils;
+import org.springframework.cloud.dataflow.server.DataFlowServerUtil;
+import org.springframework.cloud.dataflow.server.config.apps.CommonApplicationProperties;
+import org.springframework.cloud.dataflow.server.controller.WhitelistProperties;
+import org.springframework.cloud.deployer.spi.app.AppDeployer;
+import org.springframework.cloud.deployer.spi.core.AppDefinition;
+import org.springframework.cloud.deployer.spi.core.AppDeploymentRequest;
+import org.springframework.core.io.Resource;
+import org.springframework.util.Assert;
+
+import static org.springframework.cloud.deployer.spi.app.AppDeployer.COUNT_PROPERTY_KEY;
+
+/**
+ * @author Eric Bottard
+ * @author Mark Fisher
+ * @author Patrick Peralta
+ * @author Ilayaperumal Gopinathan
+ * @author Marius Bogoevici
+ * @author Janne Valkealahti
+ */
+public class StreamDeploymentPropertiesUtils {
+
+	private static Log logger = LogFactory.getLog(StreamDeploymentPropertiesUtils.class);
+
+	/**
+	 * This is the spring boot property key that Spring Cloud Stream uses to filter the
+	 * metrics to import when the specific Spring Cloud Stream "applicaiton" trigger is
+	 * fired for metrics export.
+	 */
+	private static final String METRICS_TRIGGER_INCLUDES = "spring.metrics.export.triggers.application.includes";
+
+	private static final String DEFAULT_PARTITION_KEY_EXPRESSION = "payload";
+
+	private final AppRegistry appRegistry;
+
+	private final CommonApplicationProperties commonApplicationProperties;
+
+	private final WhitelistProperties whitelistProperties;
+
+	public StreamDeploymentPropertiesUtils(AppRegistry appRegistry,
+			CommonApplicationProperties commonApplicationProperties,
+			ApplicationConfigurationMetadataResolver metadataResolver) {
+		Assert.notNull(appRegistry, "AppRegistry must not be null");
+		Assert.notNull(commonApplicationProperties, "CommonApplicationProperties must not be null");
+		Assert.notNull(metadataResolver, "MetadataResolver must not be null");
+		this.appRegistry = appRegistry;
+		this.commonApplicationProperties = commonApplicationProperties;
+		this.whitelistProperties = new WhitelistProperties(metadataResolver);
+	}
+
+	List<AppDeploymentRequest> createRequests(StreamDefinition stream,
```

Comment:
make public

Reply:
Made the private method package protected so as to be able to write finer grained tests.